pkg/cloudprovider/cloudprovider.go

```diff
@@ -337,3 +348,16 @@ func (c *CloudProvider) instanceToMachine(ctx context.Context, ec2instance *ec2.
 	machine.Status.ProviderID = fmt.Sprintf("aws:///%s/%s", aws.StringValue(ec2instance.Placement.AvailabilityZone), aws.StringValue(ec2instance.InstanceId))
 	return machine
 }
+
+func (c *CloudProvider) getInstance(ctx context.Context, id string) (*ec2.Instance, error) {
```

Comment:
Do you think this needs to be its own function call? I think we are only saving on the parsing bit here since we will still have to update the logging context and still have to call the Get() function for any callers to this function

Reply:
I'm happy to do either way. I moved it over here because you asked to https://github.com/aws/karpenter/pull/3587#discussion_r1139819209. I think saving the parsing bit makes sense.

Reply:
Oh! Sorry, I made that comment a long time ago and forgot about it. Looking at it now, I think leaving it the way that it was was fine.